### PR TITLE
Pass context not span_context to sampler in tracer

### PR DESCRIPTION
**Pull request by rdooley**
https://github.com/yangxikun/opentelemetry-lua/issues/48

### CODE REVIEW

**Comment by rdooley** [comment]
I will work on adding spec coverage for the other samplers now in a separate commit

File: lib/opentelemetry/trace/sampling/parent_base_sampler.lua [hunk 1]
```diff
@@ -21,7 +21,7 @@ function _M.new(root)
 end
 
 function _M.should_sample(self, parameters)
-    local parent_ctx = parameters.parent_ctx
+    local parent_ctx = parameters.parent_ctx:span_context()
```
Comment (yangxikun) [inline lib/opentelemetry/trace/sampling/parent_base_sampler.lua:24]: `parent_ctx` and `parameters.parent_ctx` now have different meaning, `parameters.parent_ctx` should rename to `parameters.context`

**Review by yangxikun — CHANGES_REQUESTED** [review]

**Comment by rdooley** [comment]
<details>

<summary> oddly working locally </summary>

```
 make openresty-unit-test CONTAINER_ORCHESTRATOR=podman-compose
podman-compose exec  -- openresty bash -c 'cd /opt/opentelemetry-lua && prove -r'
['podman', '--version', '']
using podman version: 4.3.0
podman exec --interactive --tty opentelemetry-lua_openresty_1 bash -c cd /opt/opentelemetry-lua && prove -r
./t/context.t ................................ ok
./t/trace/batch_span_processor.t ............. ok
./t/trace/sampling/parent_base_sampler.t ..... ok
./t/trace/sampling/trace_id_ratio_sampler.t .. ok
./t/trace/span_finish_timestamp.t ............ ok
./t/trace/tracer.t ........................... ok
./t/trace/tracer_provider.t .................. ok
All tests successful.
Files=7, Tests=41, 21 wallclock secs ( 0.02 usr  0.01 sys +  0.45 cusr  0.24 csys =  0.72 CPU)
Result: PASS
exit code: 0
```

</details>

**Comment by rdooley** [comment]
I mistakenly thought tests were passing due developer error (poor git state locally)
Apologies

**Review by yangxikun — APPROVED** [review]

**Comment by yangxikun** [comment]
image

@rdooley These need to fix.

**Comment by rdooley** [comment]
Finally clicked the button to get [workflows for my fork](https://github.com/rdooley/opentelemetry-lua/actions/runs/3465725167/jobs/5788807806) so now i have some confidence in that, and learned how to replicate this stuff locally which was a little bumpy at times.

Note to others if you see this `cd /opt/opentelemetry-lua && luarocks make && nginx -s reload` on the openresty container

**Pull request merged**